client/service_create.go

```diff
@@ -58,6 +84,27 @@ func (cli *Client) ServiceCreate(ctx context.Context, service swarm.ServiceSpec,
 	return response, err
 }
 
+func getImageWithDigest(ctx context.Context, cli *Client, image, encodedAuth string) (string, []swarm.Platform, error) {
```

Comment:
This also returns the set of platforms. Maybe `queryRegistry` or `imageDigestAndPlatforms`?